### PR TITLE
SLVS-1287 Show goldbar on `didSkipLoadingPlugin` + remove legacy locator logic for NodeJS

### PR DESCRIPTION
**Pull request by georgii-borovinskikh-sonarsource**

### CODE REVIEW

File: src/SLCore.Listeners/SLCore.Listeners.csproj [hunk 1]
```diff
@@ -9,6 +9,7 @@
 
   <ItemGroup>
     <ProjectReference Include="..\ConnectedMode\ConnectedMode.csproj" />
+    <ProjectReference Include="..\Integration\Integration.csproj" />
```
Comment (gabriela-trutan-sonarsource) [inline src/SLCore.Listeners/SLCore.Listeners.csproj:12]: I think it is a bad idea to add a dependency from SLCore.Listeners assembly to the Integration assembly, which according to this diagram (https://lucid.app/lucidchart/6905ec0c-fb12-4fc5-b0f3-37ca9cf3c997/edit?invitationId=inv_513583e6-c1e2-4cc9-9051-2f54f09c845e&page=0_0#) is used for VS integration. 

I think instead IUnsupportedNodeVersionNotificationService can be moved inside Core, then we can remove this dependency.

**Review by gabriela-trutan-sonarsource — CHANGES_REQUESTED** [review]
There are a few very minor comments, but I am more concerned about the dependency that was introduced in the SlCore.Listeners to the Integration assembly.

File: src/Integration.UnitTests/NodeJS/Locator/EnvironmentVariableNodeLocationProviderTests.cs [hunk 2]
```diff
@@ -46,17 +46,17 @@ public void Get_HasEnvironmentVariable_ReturnsListWithValue()
...
 
             var testSubject = CreateTestSubject();
             var result = testSubject.Get();
 
-            result.Should().BeEmpty();
+            result.Should().BeNull();
         }
 
-        private EnvironmentVariableNodeLocationsProvider CreateTestSubject()
+        private EnvironmentVariableNodeLocationProvider CreateTestSubject()
```
[the remark this answers is not in the document]
Comment (georgii-borovinskikh-sonarsource) [reply src/Integration.UnitTests/NodeJS/Locator/EnvironmentVariableNodeLocationProviderTests.cs:57]: These tests are not new

File: src/SLCore.Listeners/SLCore.Listeners.csproj (hunk 1 above)
Comment (georgii-borovinskikh-sonarsource) [reply src/SLCore.Listeners/SLCore.Listeners.csproj:12]: That's the whole purpose of the separate Listeners assembly, to allow us to reference these kind of assemblies. ConnectedMode.csproj already has a dependency to that.

File: src/Integration.UnitTests/NodeJS/Notifications/UnsupportedNodeVersionNotificationServiceTests.cs [hunk 3]
```diff
@@ -49,14 +54,14 @@ public void Show_ShowsCorrectMessageAndNotificationId()
...
+            testSubject.Show(language, expectedVersion, actualVersion);
 
             notificationService.Verify(x => x.ShowNotification(It.IsAny<INotification>()), Times.Once);
             notificationService.VerifyNoOtherCalls();
 
             createdNotification.Should().NotBeNull();
-            createdNotification.Id.Should().Be("sonarlint.nodejs.min.version.not.found.14.17");
-            createdNotification.Message.Should().Be(NotificationStrings.NotificationUnsupportedNode);
+            createdNotification.Id.Should().Be("sonarlint.nodejs.min.version.not.found");
+            createdNotification.Message.Should().Be($"SonarLint: {language} analysis failed. Could not find a Node.js runtime (required: >={expectedVersion}, actual: {displayActualVersion}) on your computer.");
```
[the remark this answers is not in the document]
Comment (georgii-borovinskikh-sonarsource) [reply src/Integration.UnitTests/NodeJS/Notifications/UnsupportedNodeVersionNotificationServiceTests.cs:64]: there was this thread once https://github.com/SonarSource/sonarlint-visualstudio/pull/5390#discussion_r1584690857

File: src/Integration.UnitTests/NodeJS/Locator/EnvironmentVariableNodeLocationProviderTests.cs (hunk 2 above)
[the remark this answers is not in the document]
Comment (gabriela-trutan-sonarsource) [reply src/Integration.UnitTests/NodeJS/Locator/EnvironmentVariableNodeLocationProviderTests.cs:57]: I know. But the class is fairly small, so I thought why not CaYC?

File: src/SLCore.Listeners/SLCore.Listeners.csproj (hunk 1 above)
Comment (gabriela-trutan-sonarsource) [reply src/SLCore.Listeners/SLCore.Listeners.csproj:12]: As discussed, please give it a try to move the IUnsupportedNodeVersionNotificationService to a better place, that will allow us to remove the dependency to Integration. If it's not easy, we can discuss again and I can approve the PR.

File: src/Integration.UnitTests/NodeJS/Notifications/UnsupportedNodeVersionNotificationServiceTests.cs (hunk 3 above)
[the remark this answers is not in the document]
Comment (gabriela-trutan-sonarsource) [reply src/Integration.UnitTests/NodeJS/Notifications/UnsupportedNodeVersionNotificationServiceTests.cs:64]: I disagree with the thread, but I understand your reasoning.

**Review by gabriela-trutan-sonarsource — CHANGES_REQUESTED** [review]
As discussed, please give it a try to move the IUnsupportedNodeVersionNotificationService to a better place, that will allow us to remove the dependency to Integration. If it's not easy, we can discuss again and I can approve the PR.

File: src/Integration.UnitTests/NodeJS/Locator/EnvironmentVariableNodeLocationProviderTests.cs (hunk 2 above)
[the remark this answers is not in the document]
Comment (georgii-borovinskikh-sonarsource) [reply src/Integration.UnitTests/NodeJS/Locator/EnvironmentVariableNodeLocationProviderTests.cs:57]: We don't have time for it at this point in the sprint

**Review by gabriela-trutan-sonarsource — APPROVED** [review]

**Pull request merged**